ext/intl/timezone/timezone_methods.cpp

```diff
@@ -133,53 +133,45 @@ U_CFUNC PHP_FUNCTION(intltz_get_unknown)
 U_CFUNC PHP_FUNCTION(intltz_create_enumeration)
 {
 	zval				*arg = NULL;
-	StringEnumeration	*se	  = NULL;
+	zend_string                     *args = NULL;
+	zend_long                       argl = 0;
+	StringEnumeration	        *se = NULL;
 	intl_error_reset(NULL);
 
-	/* double indirection to have the zend engine destroy the new zval that
-	 * results from separation */
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|z", &arg) == FAILURE) {
-		RETURN_THROWS();
-	}
+	ZEND_PARSE_PARAMETERS_START(0, 1)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_STR_OR_LONG(args, argl);
+	ZEND_PARSE_PARAMETERS_END();
 
-	if (arg == NULL || Z_TYPE_P(arg) == IS_NULL) {
+	if (ZEND_NUM_ARGS() == 0) {
 		se = TimeZone::createEnumeration();
-	} else if (Z_TYPE_P(arg) == IS_LONG) {
-int_offset:
-		if (UNEXPECTED(Z_LVAL_P(arg) < (zend_long)INT32_MIN ||
-				Z_LVAL_P(arg) > (zend_long)INT32_MAX)) {
-			zend_argument_value_error(1, "value is out of range");
-			RETURN_THROWS();
+	} else {
+		if (args) {
+			double dval;
+			switch (is_numeric_string(ZSTR_VAL(args), ZSTR_LEN(args), &argl, &dval, 0)) {
+			case IS_DOUBLE:
+				argl = zend_dval_to_lval(dval);
+				if (!zend_is_long_compatible(argl, dval)) {
+					zend_incompatible_double_to_long_error(dval);
+					RETURN_FALSE;
+				}
```

Comment:
Add a test for this, and I don't think you should return false here, as the behaviour should be the same, just emitting a deprecation notice :)